src/Bart/Jenkins/Job.php

```diff
@@ -9,52 +9,122 @@
  */
 class Job
 {
-	private $base_job_url;
-	private $default_params = array();
+	private $baseJobUrl;
+	private $default_params = [];
 	private $my_build_id;
 	private $metadata;
 	/** @var \Logger */
 	private $logger;
 
+	/** @var array $curlOptions */
+	private $curlOptions;
+	/** @var string $port */
+	private $port;
+
+
+
 	/**
-	 * Load metadata about a project
-	 */
-	public function __construct($domain, $job_name)
+	 * Job constructor. Loads metadata about a project.
+	 * @param string $domain The
+	 * @param string $baseProject The top-level name of your Project.
+	 * @param array $subProjects If your Jenkins Job is not defined in the top-level project,
+	 * this parameter must be passed in. For example, if your Job is defined in the following
+	 * third level project: Base->Build->Example, you must pass in 'Base' as the $baseProject
+	 * parameter, and the array ['Build', 'Example'] as the $subProjects parameter. The order
+	 * of the array is maintained when loading the metadata, and hence it's critical.
+	 * @param string $protocol 'http' or 'https'.
+	 * @param string $user User to authenticate against.
+	 * @param string $token API token corresponding to user.
+	 * NOTE: The $user and $token must be passed in if the Jenkins instance doesn't support
+	 * anonymous connections.
+	 * @param int $port The port can be generally be determined by the $protocol. 'http' corresponds
+	 * to port 8080, while 'https' corresponds to 443. If the $port is passed in, it will override
+	 * that determination.
+     */
+	public function __construct(
+		$domain,
+		$baseProject,
+		array $subProjects = [],
+		$protocol = 'http',
+		$user = null,
```

Comment:
This is a lot of arguments (>4). It might be easier to have a `setAuth()` method, or provide a `JenkinsJobBuilder` class that creates a `Job` and that class can expose setter methods, a la Effective Java section 1.